src/server/rpc/procedures/hurricane-tracking/hurricane-tracking.js

```diff
@@ -0,0 +1,133 @@
+const lineReader = require('line-reader');
+const path = require('path');
+
+const table = [
+    ['name', 'year', 'month', 'day', 'time', 'recordID', 'status', 'latitude', 'longitude', 'maxWind', 'minPressure']
+];
+
+let name = '';
+
+lineReader.eachLine(path.join(__dirname,'hurdat2-1851-2018-051019.txt'), function (line) {
+    if (line.startsWith('AL')){
+        name = line.substring(19, 28).trim();
+    } else {
+        let year = line.substring(0, 4);
+        let month = line.substring(4,6);
+        let day = line.substring(6,8);
+        let time = line.substring(10,14);
+        let recordID = line.substring(15,17);
+        let status = line.substring(18,21).trim();
+        let latitude = line.substring(23,27);
+        let longitude = '-' + line.substring(31,35);
+        let maxWind = line.substring(39,41);
+        let minPressure = line.substring(43,47).trim();
+        let data = [name, year, month, day, time, recordID, status, latitude, longitude, maxWind, minPressure];
+        table.push(data);
+    }
+});
+
+const hurricaneTracker = {};
+
+hurricaneTracker.serviceName = 'HurricaneInfo';
+
+/**
+ * returns a table displaying all of the recorded hurricanes and their respective information
+ * @returns {Array} table - displays hurricane information since 1851
+ */
+
+hurricaneTracker.getFullTable = function(){
+    return table;
+};
+
+/**
+ *
+ * @param {string} name - name of the hurricane
+ * @param {string} year - year that the hurricane occurred in
+ * @returns {array} table - table with the information for all hurricanes matching the inputted name
+ */
+
+hurricaneTracker.getHurricane = function(name, year){
+    let parsedTable = [table[0]];
+    for (let i = 1; i < table.length; i++) {
+        if (table[i][0] === name.toUpperCase() && table[i][1] === year.toString()){
+            parsedTable.push(table[i]);
+        }
+    }
+    if (parsedTable.length === 1) {
+        return 'Invalid Hurricane Name or Year';
```

Comment:
On second thought, do we even want to throw an error here? It may be better to just change the type of the `year` input to ensure the user is providing a valid value. As long as the name is a string and year is an actual year, it seems fair to just return an empty list if nothing is found. I am adding a comment above about the type used for the year.